### ISSUE TITLE
Terraform configuration file updates

### ISSUE THREAD

**Issue by Asiya-Yunusa** [Azure/azure-spring-cloud-reference-architecture#115]
**Version used:** Terraform version: 1.1.7

I am trying to deploy Spring Cloud using the terraform config files provided in this repo. I have run it previously with an older terraform version (1.1.3) and haven't gotten the following errors.

**-modules/azure_firewall/main.tf:** 
 The sku_name and sku_tier are now required firewall attributes. They need to be specified in the azure_firewall_instance resource. 
The "AZFW_Hub" sku_name does not support IP Configurations.

**-modules/my_sql/main.tf**
The minimal TLS version for the my_sql_server resource has to be specified with a valid parameter.
![image](https://user-images.githubusercontent.com/43571838/161326285-c225df13-d974-4083-9a4e-527eb3353be0.png)

**-modules/key_vault/main.tf**
The permission naming case has to updated from lower case from ["get, "regeneratekey", "listsas"] to ["Get", "RegenerateKey", "ListSAS"]

**-modules/azure_spring_cloud/main.tf**
The "instrumentation_key" argument is deprecated and should be removed.

**Comment by agoncal** [comment]
I have the same issues (using Terraform 1.1.6):

* `modules/azure_firewall/main.tf`: requires `sku-tier` and `sku_name` in` azure_firewall`
* `modules/azure_spring_cloud/main.tf`: The "instrumentation_key" argument is deprecated and should be removed
* `modules/key_vault/main.tf`: renaming permission
* `modules/my_sql/main.tf`: TLS version

**Comment by ryhud** [comment]
@Asiya-Yunusa  and @agoncal,  The version of hashicorp/azurerm in main.tf has now been set to static - 2.50. This should get past the above deployment issues. We are working on a new version of the samples which will be updated to work with the newer provider versions.

**Comment by Asiya-Yunusa** [comment]
Hello @ryhud thank you this verision works. But there's an issue with the springcloud service deployment. Deploying  of the service to UK South times out after one hour, but when I deploy to East US the service deploys without any issues. This is particular to the springcloud service, all other resources deploy in UK South.

**Comment by ryhud** [comment]
@Asiya-Yunusa, there was a region specific firewall rule within Azure Firewall. This has been resolved.

**Comment by Asiya-Yunusa** [comment]
@ryhud  thanks for the update. I will test and get back to you.

**Comment by agoncal** [comment]
I have the following errors (deploying to East US) : 

````
╷
│ Error: A resource with the ID "/subscriptions/6415dfdd-f0dd-46a5-a418-49da0a28e77a/resourcegroups/sc-corp-rg/providers/microsoft.network/azurefirewalls/fwhub-j0wsjmbvk373d/providers/microsoft.insights/diagnosticSettings/monitoring" already exists - to be managed via Terraform this resource needs to be imported into the State. Please see the resource documentation for "azurerm_monitor_diagnostic_setting" for more information.
│
│   with module.hub_spoke.module.azure_firewall.azurerm_monitor_diagnostic_setting.azfw_diag,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058333601%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Ez5Gwjibvza1XLHloaJNGxSIOqa8P%2FWMFt7V0xoOY10%3D&reserved=0) line 40, in resource "azurerm_monitor_diagnostic_setting" "azfw_diag":
│   40: resource "azurerm_monitor_diagnostic_setting" "azfw_diag" {
│
╵
╷
│ Error: creating/updating Application Rule Collection "PrivateAKSAppRules" in Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): network.AzureFirewallsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="AnotherOperationInProgress" Message="Another operation on this or dependent resource is in progress. To retrieve status of the operation use uri: [https://management.azure.com/subscriptions/6415dfdd-f0dd-46a5-a418-49da0a28e77a/providers/Microsoft.Network/locations/eastus/operations/6cdaaeb8-61ce-4f3a-bac8-e32c179ca196?api-version=2020-07-01](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmanagement.azure.com%2Fsubscriptions%2F6415dfdd-f0dd-46a5-a418-49da0a28e77a%2Fproviders%2FMicrosoft.Network%2Flocations%2Feastus%2Foperations%2F6cdaaeb8-61ce-4f3a-bac8-e32c179ca196%3Fapi-version%3D2020-07-01&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058333601%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=fJkCuToFIs%2F5p03AnmH5857EZ7SM%2BDTxb%2BcSEBZnRwc%3D&reserved=0)." Details=[]
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_application_rule_collection.private_aks,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058333601%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Ez5Gwjibvza1XLHloaJNGxSIOqa8P%2FWMFt7V0xoOY10%3D&reserved=0) line 104, in resource "azurerm_firewall_application_rule_collection" "private_aks":
│  104: resource "azurerm_firewall_application_rule_collection" "private_aks" {
│
╵
╷
│ Error: waiting for creation/update of Network Rule Collection "spring_cloud_network_tcp_rules" of Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): Future#WaitForCompletion: context has been cancelled: StatusCode=200 -- Original Error: context deadline exceeded
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_network_rule_collection.spring_cloud_tcp,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058333601%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Ez5Gwjibvza1XLHloaJNGxSIOqa8P%2FWMFt7V0xoOY10%3D&reserved=0) line 121, in resource "azurerm_firewall_network_rule_collection" "spring_cloud_tcp":
│  121: resource "azurerm_firewall_network_rule_collection" "spring_cloud_tcp" {
│
╵
╷
│ Error: creating/updating Network Rule Collection "AKS_network_tcp_rules" in Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): network.AzureFirewallsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="AnotherOperationInProgress" Message="Another operation on this or dependent resource is in progress. To retrieve status of the operation use uri: [https://management.azure.com/subscriptions/6415dfdd-f0dd-46a5-a418-49da0a28e77a/providers/Microsoft.Network/locations/eastus/operations/6cdaaeb8-61ce-4f3a-bac8-e32c179ca196?api-version=2020-07-01](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmanagement.azure.com%2Fsubscriptions%2F6415dfdd-f0dd-46a5-a418-49da0a28e77a%2Fproviders%2FMicrosoft.Network%2Flocations%2Feastus%2Foperations%2F6cdaaeb8-61ce-4f3a-bac8-e32c179ca196%3Fapi-version%3D2020-07-01&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WExpdp6q7M%2BrYrYD9Wb7RvyHumqaWGbhYNfdnQwsaNw%3D&reserved=0)." Details=[]
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_network_rule_collection.aks_netrules_tcp,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6IiiO640KS4ci0IYsIt5ofRvV9R1YHUwlLBy1QI7mF8%3D&reserved=0) line 172, in resource "azurerm_firewall_network_rule_collection" "aks_netrules_tcp":
│  172: resource "azurerm_firewall_network_rule_collection" "aks_netrules_tcp" {
│
╵
╷
│ Error: creating/updating Application Rule Collection "Microsoft_CRL_rules" in Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): network.AzureFirewallsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="AnotherOperationInProgress" Message="Another operation on this or dependent resource is in progress. To retrieve status of the operation use uri: [https://management.azure.com/subscriptions/6415dfdd-f0dd-46a5-a418-49da0a28e77a/providers/Microsoft.Network/locations/eastus/operations/5444c867-c1a8-4581-a548-97bfa867fbbb?api-version=2020-07-01](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmanagement.azure.com%2Fsubscriptions%2F6415dfdd-f0dd-46a5-a418-49da0a28e77a%2Fproviders%2FMicrosoft.Network%2Flocations%2Feastus%2Foperations%2F5444c867-c1a8-4581-a548-97bfa867fbbb%3Fapi-version%3D2020-07-01&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kyL38jYkMK7XILyBN1s4uCuPut%2BA80VmYqukMGaAd9c%3D&reserved=0)." Details=[]
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_application_rule_collection.microsoft_crl,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6IiiO640KS4ci0IYsIt5ofRvV9R1YHUwlLBy1QI7mF8%3D&reserved=0) line 229, in resource "azurerm_firewall_application_rule_collection" "microsoft_crl":
│  229: resource "azurerm_firewall_application_rule_collection" "microsoft_crl" {
│
╵
╷
│ Error: creating/updating Application Rule Collection "Microsoft_Blob_rules" in Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): network.AzureFirewallsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="AnotherOperationInProgress" Message="Another operation on this or dependent resource is in progress. To retrieve status of the operation use uri: [https://management.azure.com/subscriptions/6415dfdd-f0dd-46a5-a418-49da0a28e77a/providers/Microsoft.Network/locations/eastus/operations/6cdaaeb8-61ce-4f3a-bac8-e32c179ca196?api-version=2020-07-01](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmanagement.azure.com%2Fsubscriptions%2F6415dfdd-f0dd-46a5-a418-49da0a28e77a%2Fproviders%2FMicrosoft.Network%2Flocations%2Feastus%2Foperations%2F6cdaaeb8-61ce-4f3a-bac8-e32c179ca196%3Fapi-version%3D2020-07-01&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WExpdp6q7M%2BrYrYD9Wb7RvyHumqaWGbhYNfdnQwsaNw%3D&reserved=0)." Details=[]
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_application_rule_collection.BlobStorage_rules,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6IiiO640KS4ci0IYsIt5ofRvV9R1YHUwlLBy1QI7mF8%3D&reserved=0) line 259, in resource "azurerm_firewall_application_rule_collection" "BlobStorage_rules":
│  259: resource "azurerm_firewall_application_rule_collection" "BlobStorage_rules" {
│
╵
╷
│ Error: waiting for creation/update of Application Rule Collection "Microsoft_Login_rules" of Firewall "fwhub-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): context deadline exceeded
│
│   with module.hub_spoke.module.azure_firewall.azurerm_firewall_application_rule_collection.Microsoft_Login_rules,
│   on modules/azure_firewall/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=6IiiO640KS4ci0IYsIt5ofRvV9R1YHUwlLBy1QI7mF8%3D&reserved=0) line 390, in resource "azurerm_firewall_application_rule_collection" "Microsoft_Login_rules":
│  390: resource "azurerm_firewall_application_rule_collection" "Microsoft_Login_rules" {
│
╵
╷
│ Error: waiting for creation of Spring Cloud Service: (Spring Name "spring-j0wsjmbvk373d" / Resource Group "sc-corp-rg"): Code="BadRequest" Message="Failed to create Azure Spring Cloud service instance spring-j0wsjmbvk373d in VNet due to customer error: Required traffic is not whitelisted, refer to [https://docs.microsoft.com/en-us/azure/spring-cloud/vnet-customer-responsibilities.](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fazure%2Fspring-cloud%2Fvnet-customer-responsibilities.&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058383609%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=pm%2FcZbQHabeeLQ7RsUll969I6YoUWQGE9Ecv2cUbU3Y%3D&reserved=0)."
│
│   with [module.spring_cloud.azurerm_spring_cloud_service.sc](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmodule.spring_cloud.azurerm_spring_cloud_service.sc%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058433598%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=QhlsRF0HL88Xu0YiXStald%2FcUI7cF89zjeB3q%2Bt2S2k%3D&reserved=0),
│   on modules/azure_spring_cloud/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058433598%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=xYSToFJkZUV3fz4%2FToim9ZHaOdHOZ6VDi8q7UqLTDl0%3D&reserved=0) line 64, in resource "azurerm_spring_cloud_service" "sc":
│   64: resource "azurerm_spring_cloud_service" "sc" {
│
╵
╷
│ Error: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidParameter" Message="The Admin Username specified is not allowed." Target="adminUsername"
│
│   with module.hub_spoke.module.jump_host.azurerm_virtual_machine.jump_host,
│   on modules/jump_host/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058433598%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=xYSToFJkZUV3fz4%2FToim9ZHaOdHOZ6VDi8q7UqLTDl0%3D&reserved=0) line 40, in resource "azurerm_virtual_machine" "jump_host":
│   40: resource "azurerm_virtual_machine" "jump_host" {
│
╵
╷
│ Error: creating MySQL Server "mysql-j0wsjmbvk373d" (Resource Group "sc-corp-rg"): mysql.ServersClient#Create: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidLoginName" Message="User login name is blocked or reserved, please use another name."
│
│   with module.my_sql.azurerm_mysql_server.mysql,
│   on modules/my_sql/[main.tf](https://nam06.safelinks.protection.outlook.com/?url=http%3A%2F%2Fmain.tf%2F&data=05%7C01%7Cantoniomanug%40microsoft.com%7C4dfd653b43e8421d7afb08da1b953d3b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637852625058433598%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=xYSToFJkZUV3fz4%2FToim9ZHaOdHOZ6VDi8q7UqLTDl0%3D&reserved=0) line 34, in resource "azurerm_mysql_server" "mysql":
│   34: resource "azurerm_mysql_server" "mysql" {
```

**Comment by Asiya-Yunusa** [comment]
![image](https://user-images.githubusercontent.com/43571838/163672337-3246e750-799b-46ab-909f-7999722cabaa.png)
@ryhud I added some resources such as redis app association and a flexible mysql server. But it seems the version of the azurerm provider doesn't support them. Can I suggest making the changes that led to the initial errors mentioned in the first comment, because I am able to create them with that version.

**Comment by kriation** [comment]
Fixed as of PR #131
